src/services/cookie.service.js

```diff
@@ -0,0 +1,54 @@
+'use strict'
+
+// Used for generating a session id which is saved as a cookie
+const uuid4 = require('uuid/v4')
+
+const Constants = require('../constants')
+const ServerLoggingService = require('../services/serverLogging.service')
+const ActiveDirectoryAuthService = require('../services/activeDirectoryAuth.service')
+const authService = new ActiveDirectoryAuthService()
+
+module.exports = class CookieService {
+  static async generateCookie (reply) {
```

Comment:
As discussed, now you've gone to all that effort to create an Application in Dynamics, it's its ID we need to pass in rather than generate one.

Reply:
Well spotted - I have removed the generation of a UUID from the app as it was being over-written in any case by the Dynamics application ID once the application model object was saved.